index.d.ts

```diff
@@ -127,9 +127,9 @@ export interface FastifyDynamicSwaggerOptions extends FastifySwaggerOptions {
    */
   stripBasePath?: boolean;
   /**
-   * Overwrite the route schema
+   * custom function to transform the route's schema and url
    */
-  transform?: Function;
+  transform?: <S = any>({schema, url}: {schema: S, url: string}) => { schema: S, url: string };
```

Comment:
```suggestion
  transform?: <S = any>({schema, url}: {schema: S, url: string}) => { schema: JSONObject, url: string };
```

@climba03003 I would go with `unknown` instead of `any` for a user input WDYT ?

Reply:
My intention here is not forcing the user to specify a type and they are good to go. If it is `= unknown`, that means the user is force to specify the type, then I would go for remove the `= unknown`.

Reply:
S could be `FastifySchema` (by either direct ref or copy paste the type into this lib) :

https://github.com/fastify/fastify/blob/596e7cad346fa31171f26260a9612b20f4646bcc/types/schema.d.ts#L9-L15

is it a good fit?

Reply:
Yes, it is a possible change. 
```
<S extends FastifySchema = FastifySchema>({schema, url}: {schema: S, url: string}) => { schema: JSONObject, url: string }
```

Reply:
This one is better IMO. ^^

Reply:
done and it passes the tests 🥳 

I also squashed the commits and force pushed again to make history a lil nicer.